### PR TITLE
Functional version

### PR DESCRIPTION
**Pull request by MartinaSevero**
# Functional version

### 📑 Description:

- Added Random Forest, Gradient Boosting, Decision trees, Logistic Regression and Neural Networks algorithms.
- Added search for best hyperparameters.

### CODE REVIEW

**Review by SantiagoTonarelli — APPROVED** [review]
LGTM
